Source/SelfService/Web/microservice/microserviceView/configuration/environmentVariablesSection/environmentVariableSection.tsx

```diff
@@ -0,0 +1,288 @@
+        // if (JSON.stringify(envVariableTableRows) === JSON.stringify(updatedRow)) {
+        //     enqueueSnackbar('No changes detected', { variant: 'info' });
+        //     return;
+        // }
+    };
+
+    const processRowUpdate = async (newRow: GridRowModel) => {
+        setDisableAddButton(false);
+
+        const updatedRow = { ...newRow, isNew: false };
+
+        if (!validateEnvVariable(updatedRow)) {
+            handleInvalidValues(updatedRow.id);
+            return;
+        }
+
+        const updatedEnvVariable = envVariableTableRows.map(row => (row.id === newRow.id ? updatedRow : row));
+
+        const result = await updateEnvironmentVariables(applicationId, environment, microserviceId, updatedEnvVariable);
+
+        if (result) {
+            setEnvVariableTableRows(updatedEnvVariable);
+            setRestartInfoBoxIsOpen(true);
+            enqueueSnackbar(`Environment variable ${newRow.isNew ? 'added' : 'updated'}.`);
+        } else {
+            enqueueSnackbar('Could not update environment variable.', { variant: 'error' });
+        }
+
+        return updatedRow;
+    };
+
+    const handleEnvVariableAdd = () => {
+        setDisableAddButton(true);
+
+        const randomId = crypto.randomUUID();
+
+        const newEnvVariable = {
+            id: randomId,
+            name: '',
+            value: '',
+            isSecret: false,
+            isNew: true
+        };
+
+        const updateChangedRows = [...envVariableTableRows, newEnvVariable];
+        setEnvVariableTableRows(updateChangedRows);
+
+        setRowMode(prevRowMode => ({
+            ...prevRowMode,
+            [randomId]: { mode: GridRowModes.Edit, fieldToFocus: 'name' },
+        }));
+    };
+
+    const handleEnvVariableDelete = async () => {
+        const remainingEnvVariables = envVariableTableRows.filter(envVariable => !selectedRowIds.includes(envVariable.id));
+
+        const result = await updateEnvironmentVariables(applicationId, environment, microserviceId, remainingEnvVariables);
+
+        if (result) {
+            envVariableTableRows.filter(envVariable => {
+                if (selectedRowIds.includes(envVariable.id)) {
+                    enqueueSnackbar(`'${envVariable.name}' variable has been deleted.`);
+                }
+            });
+
+            setEnvVariableTableRows(remainingEnvVariables);
+            // TODO: Open info box that the microservice needs to be restarted.
+        } else {
+            enqueueSnackbar('File not deleted. Please try again.', { variant: 'error' });
+        }
+    };
+
+    // TODO: This is reused. consider moving
+    const configMapPrefix = `${environment.toLowerCase()}-${microserviceName.toLowerCase()}`;
+
+    const handleSecretEnvVariableDownload = () => {
+        const secretName = `${configMapPrefix}-secret-env-variables`;
+        const href = `${getServerUrlPrefix()}/live/application/${applicationId}/secret/${secretName}?download=1&fileType=yaml`;
```

Comment:
I would consider moving this to the `api`-layer. Then you can also move the `configmapPrefix` there as a const / helper